### PR TITLE
added notable files log to checkpoint log, added unit tests

### PR DESCRIPTION
**Pull request by James-Pickett**
* Added checkpoint log for "notable" files
* Broke existing CheckPoint code up into functions and added unit tests

### CODE REVIEW

File: pkg/log/checkpoint/checkpoint_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	mock.Mock
+}
+
+func (m *mockLogger) Log(keyvals ...interface{}) error {
+	m.Called(keyvals...)
+	return nil
+}
+
+func TestLogCheckPoint(t *testing.T) {
+	mockLogger := new(mockLogger)
```
Comment (directionless) [inline pkg/log/checkpoint/checkpoint_test.go:19]:
```suggestion
func TestLogCheckPoint(t *testing.T) {
         t.Parallel()

         mockLogger := new(mockLogger)
```

File: pkg/log/checkpoint/files.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+				results = append(results, fmt.Sprintf("%s/%s", dirname, file.Name()))
+			}
+		}
+	}
+
+	return results
+}
+
+// helper method to generate empty dir message, makes testing easier
+func emptyDirMsg(dirname string) string {
```
Comment (directionless) [inline pkg/log/checkpoint/files.go:30]:
```suggestion
// emptyDirMsg is a helper method to generate empty dir message, makes testing easier
func emptyDirMsg(dirname string) string {
```

go doc style

File: pkg/log/checkpoint/files_test.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+	"github.com/kolide/kit/ulid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFilesFound(t *testing.T) {
+	t.Parallel()
+
+	tempDir, err := os.MkdirTemp("", "log-checkpoint-files-test")
+	require.NoError(t, err, "making temp dir")
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:20]:
```suggestion
	tempDir, err := os.MkdirTemp("", "log-checkpoint-files-test")
	require.NoError(t, err, "making temp dir")
	defer os.RemoveAll(tempDir)
```

File: pkg/log/checkpoint/files_test.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+			require.NoError(t, err, "creating test files")
+
+			foundPaths := fileNamesInDirs(dirs...)
+			sort.Strings(foundPaths)
+			require.Equal(t, expectedPaths, foundPaths)
+			require.Equal(t, tt.dirsToCreate*tt.filesPerDir, len(foundPaths))
+		})
+	}
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:42]: I think better to do this as a defer above. Doing this here means it won't trigger if a test fails. Also, if os.RemoveAll fails, no point in failing the testing.

File: pkg/log/checkpoint/files_test.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+func TestDirNotFound(t *testing.T) {
+	t.Parallel()
+
+	tempDir, err := os.MkdirTemp("", "log-checkpoint-files-test")
+	require.NoError(t, err, "making temp dir")
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:49]:
```suggestion
	tempDir, err := os.MkdirTemp("", "log-checkpoint-files-test")
	require.NoError(t, err, "making temp dir")
	defer os.RemoveAll(tempDir)
```

File: pkg/log/checkpoint/files_test.go [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+		expectedOutput = append(expectedOutput, pathErr.Error())
+	}
+
+	foundPaths := fileNamesInDirs(nonExistantDirs...)
+
+	require.Equal(t, expectedOutput, foundPaths)
+	require.Equal(t, len(nonExistantDirs), len(foundPaths))
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:80]: As above

File: pkg/log/checkpoint/files_test.go [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+func TestDirNotFound(t *testing.T) {
+	t.Parallel()
+
+	tempDir, err := os.MkdirTemp("", "log-checkpoint-files-test")
+	require.NoError(t, err, "making temp dir")
+
+	const pathFmt = "%s/%s"
+	nonExistantDirs := []string{
+		fmt.Sprintf(pathFmt, tempDir, ulid.New()),
+		fmt.Sprintf(pathFmt, tempDir, ulid.New()),
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:54]: I don't think this is guaranteed to be cross platform. I'd use `filepath.Join(tempDir, ulid.New())`

File: pkg/log/checkpoint/files_test.go [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+	require.Equal(t, len(dirs), len(foundPaths))
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
+}
+
+func createTestFiles(baseDir string, dirCount int, filesPerDir int) (dirs []string, files []string, err error) {
+	files = []string{}
+
+	for i := 0; i < dirCount; i++ {
+		dir, err := os.MkdirTemp(baseDir, "notable-files-unit-test")
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:109]: If dirCount is > 1, won't they step on eachother?

File: pkg/log/checkpoint/files_test.go [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+			sort.Strings(foundPaths)
+			require.Equal(t, expectedPaths, foundPaths)
+			require.Equal(t, tt.dirsToCreate*tt.filesPerDir, len(foundPaths))
+		})
+	}
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
+}
+
+func TestDirNotFound(t *testing.T) {
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:45]: Is this the same case as `dirsToCreate: 0` above?

File: pkg/log/checkpoint/files_test.go [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+	foundPaths := fileNamesInDirs(nonExistantDirs...)
+
+	require.Equal(t, expectedOutput, foundPaths)
+	require.Equal(t, len(nonExistantDirs), len(foundPaths))
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
+}
+
+func TestDirEmpty(t *testing.T) {
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:83]: Is this the same case as `dirsToCreate: 2, filesPerDir: 0` above?

File: pkg/log/checkpoint/files_test.go (hunk 8 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files_test.go:109]: `os.MkdirTemp()` adds random string at the end of the pattern (second arg)

File: pkg/log/checkpoint/files_test.go (hunk 9 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files_test.go:45]: no, the test case where `dirsToCreate:0` is passing any empty array to the `fileNamesInDirs` function (it's not searching any dirs) where as this one is looking for a directory that does not exist. left a comment and updated variable name for clarity

File: pkg/log/checkpoint/files_test.go (hunk 10 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files_test.go:83]: I dropped the `dirsToCreate:2, filesPerDir:0` case since now we are returning something when we search an empty dir, I could re add that case to this test method, but would have to add some logic to check for empty dirs and generate expect output ... felt cleaner to break out to it's own function

File: pkg/log/checkpoint/checkpoint_test.go (hunk 1 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/checkpoint_test.go]: updated

File: pkg/log/checkpoint/files.go (hunk 2 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files.go]: updated

File: pkg/log/checkpoint/files_test.go (hunk 3 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files_test.go:21]: updated

File: pkg/log/checkpoint/files_test.go (hunk 4 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files_test.go]: updated

File: pkg/log/checkpoint/files_test.go (hunk 5 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files_test.go:50]: updated

File: pkg/log/checkpoint/files_test.go (hunk 6 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files_test.go]: updated

File: pkg/log/checkpoint/files_test.go (hunk 7 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files_test.go]: updated

File: pkg/log/checkpoint/files_test.go (hunk 8 above)
Comment (directionless) [reply pkg/log/checkpoint/files_test.go:108]: I totally forgot. Thanks!

File: pkg/log/checkpoint/files.go [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+		files, err := os.ReadDir(dirname)
+
+		switch {
+		case err != nil:
+			results = append(results, err.Error())
+		case len(files) == 0:
+			results = append(results, emptyDirMsg(dirname))
+		default:
+			for _, file := range files {
+				results = append(results, fmt.Sprintf("%s/%s", dirname, file.Name()))
```
Comment (directionless) [inline pkg/log/checkpoint/files.go:21]: `filepath.Join()` though for a string log it's a pretty small nit.

File: pkg/log/checkpoint/files_test.go [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+
+	for _, tt := range tests {
+		t.Run("testFilesFound", func(t *testing.T) {
+			dirsToSearch, expectedPaths, err := createTestFiles(tempDir, tt.dirsToCreate, tt.filesPerDir)
+			require.NoError(t, err, "creating test files")
+
+			foundPaths := fileNamesInDirs(dirsToSearch...)
+			sort.Strings(foundPaths)
+			require.Equal(t, expectedPaths, foundPaths)
+			require.Equal(t, tt.dirsToCreate*tt.filesPerDir, len(foundPaths))
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:41]: This feels unneeded with the prior test?

File: pkg/log/checkpoint/files_test.go [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+	for _, dir := range dirs {
+		expectedOutput = append(expectedOutput, emptyDirMsg(dir))
+	}
+
+	foundPaths := fileNamesInDirs(dirs...)
+
+	require.Equal(t, expectedOutput, foundPaths)
+	require.Equal(t, len(dirs), len(foundPaths))
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:101]: Should be in the defer above

File: pkg/log/checkpoint/files.go (hunk 11 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files.go]: updated

File: pkg/log/checkpoint/files_test.go (hunk 13 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files_test.go]: updated

**Review by directionless — DISMISSED** [review]

**Review by directionless — APPROVED** [review]
